spec/core/matchers/matchersUtilSpec.js

```diff
@@ -93,6 +93,24 @@ describe("matchersUtil", function() {
       expect(jasmineUnderTest.matchersUtil.equals(new Error("foo"), new Error("bar"))).toBe(false);
     });
 
+    it("fails for objects with different constructors", function() {
+      function One() {}
+      function Two() {}
+
+      expect(jasmineUnderTest.matchersUtil.equals(new One(), new Two())).toBe(false);
+    });
+
+    if (typeof document === 'object') {
+      it("passes for equivalent objects from different frames", function() {
```

Comment:
Whoops—I missed those other specs because I wasn't running the tests in Node on a regular basis (see: #1160).